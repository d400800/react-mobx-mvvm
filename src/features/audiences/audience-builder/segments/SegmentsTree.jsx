import React from 'react';

import {Box, Grid, Typography} from '@material-ui/core';

const SegmentsTree = ({segmentGroup, segmentGroupIndex}) => {
    function shouldPrintOperator(segmentGroupIndex, i) {
        // omit operator if it is the 1st child of the 1st parent
        return !(segmentGroupIndex === 0 && i === 0);
    }

    return (
        <Box>
            {segmentGroup.map((segment, i) => (
                <Box key={segment.id} mb={1}>
                    <Grid container spacing={3} alignItems="flex-end">
                        <Grid item xs={1}>
                            <Box color="primary.main" mr={2}>
                                {shouldPrintOperator(segmentGroupIndex, i) &&
                                    <Typography variant="subtitle2">
                                        {i === 0 ? 'or' : 'and'}
                                    </Typography>
                                }
                            </Box>
                        </Grid>

                        <Grid item xs={11}>
                            <Typography variant="h6">{segment.name}</Typography>
                        </Grid>
                    </Grid>
                </Box>
            ))}
        </Box>
    );
};

export default SegmentsTree;