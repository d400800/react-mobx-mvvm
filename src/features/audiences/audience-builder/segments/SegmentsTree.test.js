import {shallow, render} from 'enzyme';
import React from "react";

import SegmentsTree from "./SegmentsTree";

const segmentGroup = [
    {
        id: "C2",
        parentId: "C",
        name: "Click",
        numOfUsers: 13
    },
    {
        id: "C7",
        parentId: "C",
        name:"Abandoned cart",
        numOfUsers: 8
    }
];

describe('SegmentsTree enzyme shallow', () => {
    it('should match the snapshot', () => {
        const component = shallow(<SegmentsTree segmentGroup={segmentGroup} segmentGroupIndex={0} />);

        expect(component).toMatchSnapshot();
    });

    it('should not print the operator of the 1st child of the 1st parent', () => {
        const wrapper = render(<SegmentsTree segmentGroup={segmentGroup} segmentGroupIndex={0} />);

        expect(wrapper.find('[operator-box="true"]').first().text())
            .toEqual('');
    });
});