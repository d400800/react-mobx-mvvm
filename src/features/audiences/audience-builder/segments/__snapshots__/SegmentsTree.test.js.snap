// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentsTree enzyme shallow should match the snapshot 1`] = `
<Styled(MuiBox)>
  <Styled(MuiBox)
    key="C2"
    mb={1}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={1}
      >
        <Styled(MuiBox)
          color="primary.main"
          mr={2}
          operator-box="true"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={11}
      >
        <WithStyles(ForwardRef(Typography))
          id="C2-text"
          variant="h6"
        >
          Click
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    key="C7"
    mb={1}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={1}
      >
        <Styled(MuiBox)
          color="primary.main"
          mr={2}
          operator-box="true"
        >
          <WithStyles(ForwardRef(Typography))
            variant="subtitle2"
          >
            and
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={11}
      >
        <WithStyles(ForwardRef(Typography))
          id="C7-text"
          variant="h6"
        >
          Abandoned cart
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
